Задание 1 (на сопоставление)

In [ ]:
1 - E, 2 - B, 3 - F, 4 - C, 5 - A

Задание 2 (про порошок)

In [ ]:
'''Ваш ответ на задание №2
- верно указана применяемая статистика и обоснование ее применения
- получено расчетное значение статистики
- получено критическое значение статистики
- получено значение p_value
- сделан статистический вывод на основе сравнения критического значения с расчетным
- сделан статистический вывод на основе сравнения p-value и уровня значимости
- сделан содержательный вывод по поставленной проблеме'''

In [1]:
import math
from scipy.stats import norm

# Данные задачи
mu0 = 3.00        # Предполагаемое среднее (гипотеза H0)
x_bar = 3.005      # Выборочное среднее
sigma = 0.015       # Известное стандартное отклонение генеральной совокупности
n = 50              # Объем выборки
alpha = 0.05        # Уровень значимости

# Расчет стандартной ошибки среднего
SE = sigma / math.sqrt(n)

# Расчет Z-статистики
z_stat = (x_bar - mu0) / SE

# Критические значения для двустороннего теста при alpha = 0.05
z_crit = norm.ppf(1 - alpha/2)  # квантиль для верхней границы
# нижняя граница будет -z_crit

# Расчет p-value
# Для двустороннего теста: p-value = 2 * P(Z > |z_stat|)
p_value = 2 * (1 - norm.cdf(abs(z_stat)))

# Вывод результатов
print("Z-статистика:", z_stat)
print("Критическое значение Z:", z_crit)
print("p-value:", p_value)
print("Уровень значимости:", alpha)

# Статистические выводы
if abs(z_stat) > z_crit:
    print("Отвергаем H0 на основании сравнения с критическим значением.")
else:
    print("Нет оснований отвергать H0 на основании сравнения с критическим значением.")

if p_value < alpha:
    print("Отвергаем H0 на основании сравнения p-value и уровня значимости.")
else:
    print("Нет оснований отвергать H0 на основании сравнения p-value и уровня значимости.")

# Содержательный вывод
# Если H0 отвергается - значит среднее значительно отличается от 3.00 кг
if p_value < alpha:
    print("Содержательный вывод: истинное среднее значение отличается от 3.00 кг, машина нуждается в проверке/настройке.")
else:
    print("Содержательный вывод: нет статистически значимых доказательств отличия среднего от 3.00 кг, настройка не требуется.")








Z-статистика: 2.3570226039551083
Критическое значение Z: 1.959963984540054
p-value: 0.01842212545410149
Уровень значимости: 0.05
Отвергаем H0 на основании сравнения с критическим значением.
Отвергаем H0 на основании сравнения p-value и уровня значимости.
Содержательный вывод: истинное среднее значение отличается от 3.00 кг, машина нуждается в проверке/настройке.


*   **Z-статистика:** 2.3570226039551083
*   **Критическое значение Z:** 1.959963984540054
*   **p-value:** 0.01842212545410149
*   **Уровень значимости:** 0.05
Отвергаем H0 на основании сравнения с критическим значением.
Отвергаем H0 на основании сравнения p-value и уровня значимости.

**Содержательный вывод:** истинное среднее значение отличается от 3.00 кг, машина нуждается в проверке/настройке.

Задание 3 (про двигатели)

In [ ]:
'''Ваш ответ на задание №3
- верно указана применяемая статистика и обоснование ее применения
- получено расчетное значение статистики
- получено критическое значение статистики
- получено значение p_value
- сделан статистический вывод на основе сравнения критического значения с расчетным
- сделан статистический вывод на основе сравнения p-value и уровня значимости
- сделан содержательный вывод по поставленной проблеме'''

In [5]:
import math
from scipy.stats import t

# Данные
n1 = 15
mean1 = 90.0
s1 = 2.3  # стандартное отклонение в первой группе

n2 = 10
mean2 = 88.0
s2 = 2.2  # стандартное отклонение во второй группе

alpha = 0.05

# Нулевая гипотеза H0: mu1 = mu2
# Альтернативная гипотеза H1: mu1 != mu2
# Используем двухвыборочный t-тест для независимых выборок с предположением равенства дисперсий.

# Оценка дисперсий
s1_sq = s1**2
s2_sq = s2**2

# Общая (пулированная) дисперсия (поскольку объёмы сравнительно небольшие, предполагаем равенство дисперсий)
df = n1 + n2 - 2  # степени свободы
sp_sq = ((n1 - 1)*s1_sq + (n2 - 1)*s2_sq) / df
sp = math.sqrt(sp_sq)

# Стандартная ошибка разности средних
SE = sp * math.sqrt(1/n1 + 1/n2)

# t-статистика
t_stat = (mean1 - mean2) / SE

# Критическое значение t для двустороннего теста при уровне значимости alpha
t_crit = t.ppf(1 - alpha/2, df)

# p-value (двухстороннее)
p_value = 2 * (1 - t.cdf(abs(t_stat), df))

print("t-статистика:", t_stat)
print("Степени свободы:", df)
print("Критическое значение t при α=0.05:", t_crit)
print("p-value:", p_value)

# Статистический вывод на основании критического значения
if abs(t_stat) > t_crit:
    print("Отвергаем H0 на основании критического значения t.")
else:
    print("Нет оснований отвергать H0 на основании критического значения t.")

# Статистический вывод на основании p-value
if p_value < alpha:
    print("Отвергаем H0 на основании p-value.")
else:
    print("Нет оснований отвергать H0 на основании p-value.")

# Содержательный вывод
if p_value < alpha:
    print("Содержательный вывод: Есть статистически значимая разница между средними прочностными характеристиками веществ.")
    print("Следовательно, выбор может быть сделан в пользу того вещества, у которого прочностные характеристики предпочтительнее.")
else:
    print("Содержательный вывод: Нет статистически значимой разницы между средними прочностными характеристиками веществ при α=5%.")

t-статистика: 2.1663516354146894
Степени свободы: 23
Критическое значение t при α=0.05: 2.0686576104190406
p-value: 0.04089279045167049
Отвергаем H0 на основании критического значения t.
Отвергаем H0 на основании p-value.
Содержательный вывод: Есть статистически значимая разница между средними прочностными характеристиками веществ.
Следовательно, выбор может быть сделан в пользу того вещества, у которого прочностные характеристики предпочтительнее.


*  **t-статистика:** 2.1663516354146894
*  **Степени свободы:** 23
*  **Критическое значение t при α=0.05:** 2.0686576104190406
*  **p-value:** 0.04089279045167049
Отвергаем H0 на основании критического значения t.
Отвергаем H0 на основании p-value.

**Содержательный вывод:** Текст, выделенный полужирным шрифтомЕсть статистически значимая разница между средними прочностными характеристиками веществ.

**Следовательно, выбор может быть сделан в пользу того вещества, у которого прочностные характеристики предпочтительнее**

Задание 4 (про токи)

In [ ]:
'''Ваш ответ на задание №4
- верно указана применяемая статистика и обоснование ее применения
- получено расчетное значение статистики
- получено критическое значение статистики
- получено значение p_value
- сделан статистический вывод на основе сравнения критического значения с расчетным
- сделан статистический вывод на основе сравнения p-value и уровня значимости
- сделан содержательный вывод по поставленной проблеме'''

In [8]:
import math
from scipy.stats import t

# Данные для двух выборок (примерные данные)
# Здесь следует подставить фактические данные старой и новой схемы.
old_scheme = [80.1, 82.3, 84.1, 82.6, 85.3, 81.3, 83.2, 81.7, 82.2, 81.4, 81.0, 82.0]
new_scheme = [80.7, 81.3, 84.6, 81.7, 86.3, 84.3, 83.7, 84.7, 82.8, 84.4, 85.2, 84.9]

alpha = 0.05

# Нулевая гипотеза H0: mu_old = mu_new
# Альтернативная гипотеза H1: mu_old != mu_new

n_old = len(old_scheme)
n_new = len(new_scheme)

mean_old = sum(old_scheme)/n_old
mean_new = sum(new_scheme)/n_new

# Оценим выборочные дисперсии
var_old = sum((x - mean_old)**2 for x in old_scheme) / (n_old - 1)
var_new = sum((y - mean_new)**2 for y in new_scheme) / (n_new - 1)

# Предполагаем равенство дисперсий, используем пулированную дисперсию
df = n_old + n_new - 2
sp = math.sqrt(((n_old - 1)*var_old + (n_new - 1)*var_new) / df)

# Стандартная ошибка разности средних
SE = sp * math.sqrt(1/n_old + 1/n_new)

# t-статистика
t_stat = (mean_old - mean_new) / SE

# Критическое значение t для двустороннего теста при уровне значимости alpha
t_crit = t.ppf(1 - alpha/2, df)

# Расчет p-value для двустороннего теста
p_value = 2 * (1 - t.cdf(abs(t_stat), df))

print("t-статистика:", t_stat)
print("Степени свободы:", df)
print("Критическое значение t (для α=0.05):", t_crit)
print("p-value:", p_value)

# Статистический вывод на основе критического значения
if abs(t_stat) > t_crit:
    print("Отвергаем H0 на основании сравнения с критическим значением.")
else:
    print("Нет оснований отвергать H0 на основании сравнения с критическим значением.")

# Статистический вывод на основе p-value
if p_value < alpha:
    print("Отвергаем H0 на основании сравнения p-value и уровня значимости.")
else:
    print("Нет оснований отвергать H0 на основании сравнения p-value и уровня значимости.")

# Содержательный вывод
if p_value < alpha:
    print("Содержательный вывод: Есть статистически значимая разница между средними значениями.")
    print("Производитель может рассмотреть замену старой конструкции на новую.")
else:
    print("Содержательный вывод: Нет статистически значимой разницы между средними значениями.")
    print("Нет статистических оснований для замены старой конструкции на новую.")

t-статистика: -2.254474824978757
Степени свободы: 22
Критическое значение t (для α=0.05): 2.0738730679040147
p-value: 0.03445862574483893
Отвергаем H0 на основании сравнения с критическим значением.
Отвергаем H0 на основании сравнения p-value и уровня значимости.
Содержательный вывод: Есть статистически значимая разница между средними значениями.
Производитель может рассмотреть замену старой конструкции на новую.


*  t-статистика: -2.254474824978757
*  Степени свободы: 22
*  Критическое значение t (для α=0.05): 2.0738730679040147
*  p-value: 0.03445862574483893
*  Отвергаем H0 на основании сравнения с критическим значением.
*  Отвергаем H0 на основании сравнения p-value и уровня значимости.
*  Содержательный вывод: Есть статистически значимая разница между средними значениями.
Производитель может рассмотреть замену старой конструкции на новую.


Задание 5 (про глюкозу)

In [ ]:
'''Ваш ответ на задание №5
- верно указана применяемая статистика и обоснование ее применения
- получено расчетное значение статистики
- получено критическое значение статистики
- получено значение p_value
- сделан статистический вывод на основе сравнения критического значения с расчетным
- сделан статистический вывод на основе сравнения p-value и уровня значимости
- сделан содержательный вывод по поставленной проблеме'''

In [9]:
import math
from scipy.stats import t

# Данные уровней глюкозы до и после приема препарата:
before = [9.6, 8.1, 8.8, 7.9, 9.2, 8.0, 8.4, 10.1, 7.8, 8.1]
after =  [7.7, 6.2, 7.4, 7.5, 8.3, 6.2, 8.1, 8.9, 7.4, 7.0]

alpha = 0.05  # уровень значимости, соответствует доверительному интервалу 0.95

# Нулевая гипотеза H0: Средний уровень глюкозы до и после не отличается (μ_before = μ_after)
# Альтернативная гипотеза H1: Средние уровни отличаются (μ_before ≠ μ_after)

# Поскольку данные парные (одни и те же пациенты), используем парный t-тест для зависимых выборок.

# Рассчитаем разности:
differences = [b - a for b, a in zip(before, after)]

n = len(differences)
mean_diff = sum(differences)/n
# Оценим стандартное отклонение разностей
var_diff = sum((d - mean_diff)**2 for d in differences) / (n - 1)
sd_diff = math.sqrt(var_diff)

# Статистика t:
# t = (mean_diff - 0) / (sd_diff / sqrt(n))
t_stat = mean_diff / (sd_diff / math.sqrt(n))

# Число степеней свободы:
df = n - 1

# Критическое значение t при двухстороннем тесте и alpha=0.05
t_crit = t.ppf(1 - alpha/2, df)

# p-value для двустороннего теста:
p_value = 2*(1 - t.cdf(abs(t_stat), df))

print("t-статистика:", t_stat)
print("Степени свободы (df):", df)
print("Критическое значение t (α=0.05):", t_crit)
print("p-value:", p_value)

# Статистический вывод на основе сравнения с критическим значением
if abs(t_stat) > t_crit:
    print("Отвергаем H0 на основании сравнения с критическим значением.")
else:
    print("Нет оснований отвергать H0 на основании сравнения с критическим значением.")

# Статистический вывод на основе p-value
if p_value < alpha:
    print("Отвергаем H0 на основании p-value.")
else:
    print("Нет оснований отвергать H0 на основании p-value.")

# Содержательный вывод
if p_value < alpha:
    print("Содержательный вывод: Есть статистически значимый эффект от приема препарата (уровень глюкозы изменился).")
else:
    print("Содержательный вывод: Статистически значимых доказательств изменения уровня глюкозы нет.")

t-статистика: 5.713032381721968
Степени свободы (df): 9
Критическое значение t (α=0.05): 2.2621571628540993
p-value: 0.00028949403967537535
Отвергаем H0 на основании сравнения с критическим значением.
Отвергаем H0 на основании p-value.
Содержательный вывод: Есть статистически значимый эффект от приема препарата (уровень глюкозы изменился).


*  t-статистика: 5.713032381721968
*  Степени свободы (df): 9
*  Критическое значение t (α=0.05): 2.2621571628540993
*  p-value: 0.00028949403967537535
*  Отвергаем H0 на основании сравнения с критическим значением.
*  Отвергаем H0 на основании p-va

**Вывод:** Есть статистически значимый эффект от приема препарата (уровень глюкозы изменился).

Задание 6 (про тросы)

In [ ]:
'''Ваш ответ на задание №6
- верно указана применяемая статистика и обоснование ее применения
- получено расчетное значение статистики
- получено критическое значение статистики
- получено значение p_value
- сделан статистический вывод на основе сравнения критического значения с расчетным
- сделан статистический вывод на основе сравнения p-value и уровня значимости
- сделан содержательный вывод по поставленной проблеме'''

In [11]:
import math
from scipy.stats import ttest_ind, t

# Данные
new_cable = [92.7, 91.6, 94.7, 93.7, 96.5, 94.3, 93.7, 96.8, 98.9, 99.9]
old_cable = [90.1, 92.2, 94.3, 92.1, 93.4, 91.1, 93.2, 91.5, 93.2, 91.5]
alpha = 0.01

# Нулевая гипотеза H0: mu_new = mu_old
# Альтернативная гипотеза H1: mu_new != mu_old

# Welch's t-test, так как предполагается, что дисперсии не равны
t_stat, p_value = ttest_ind(new_cable, old_cable, equal_var=False)

# Для критического значения вычислим примерную степень свободы по формуле Уэлча
n1, n2 = len(new_cable), len(old_cable)
mean1, mean2 = sum(new_cable)/n1, sum(old_cable)/n2
var1 = sum((x - mean1)**2 for x in new_cable)/(n1-1)
var2 = sum((y - mean2)**2 for y in old_cable)/(n2-1)

# Степени свободы по формуле Уэлча
df = (var1/n1 + var2/n2)**2 / ((var1**2)/(n1**2*(n1-1)) + (var2**2)/(n2**2*(n2-1)))

# Критическое значение t при уровне значимости 1% (двусторонний тест)
t_crit = t.ppf(1 - alpha/2, df)

# Печатаем результат в соответствии с заданными критериями:
print("Применяемая статистика: Welch’s t-test для двух независимых выборок без предположения о равенстве дисперсий.")
print(f"Расчетная t-статистика: {t_stat:.2f}")
print(f"Степени свободы (приблизительно): {df:.2f}")
print(f"Критическое значение t (для α=0.01): ±{t_crit:.2f}")
print(f"p-value: {p_value:.3f}")

# Сравнение с критическим значением
if abs(t_stat) > t_crit:
    crit_conclusion = "Отвергаем H0 на основании критического значения."
else:
    crit_conclusion = "Нет оснований отвергать H0 на основании критического значения."

# Сравнение p-value и уровня значимости
if p_value < alpha:
    pval_conclusion = "Отвергаем H0 на основании p-value."
else:
    pval_conclusion = "Нет оснований отвергать H0 на основании p-value."

# Содержательный вывод
if p_value < alpha:
    final_conclusion = ("При уровне значимости 1% есть статистически значимые различия, "
                        "следовательно, производитель может заменить старый кабель на новый.")
else:
    final_conclusion = ("При уровне значимости 1% статистически значимых различий не обнаружено, "
                        "производитель не имеет достаточных оснований для замены старого кабеля на новый.")

print(crit_conclusion)
print(pval_conclusion)
print(final_conclusion)

Применяемая статистика: Welch’s t-test для двух независимых выборок без предположения о равенстве дисперсий.
Расчетная t-статистика: 3.22
Степени свободы (приблизительно): 12.82
Критическое значение t (для α=0.01): ±3.02
p-value: 0.007
Отвергаем H0 на основании критического значения.
Отвергаем H0 на основании p-value.
При уровне значимости 1% есть статистически значимые различия, следовательно, производитель может заменить старый кабель на новый.


Применяемая статистика: Welch’s t-test для двух независимых выборок без предположения о равенстве дисперсий.
* Расчетная t-статистика: 3.22
* Степени свободы (приблизительно): 12.82
* Критическое значение t (для α=0.01): ±3.02
* p-value: 0.007
* Отвергаем H0 на основании критического значения.
* Отвергаем H0 на основании p-value.
* При уровне значимости 1% есть статистически значимые различия, следовательно, производитель может заменить старый кабель на новый.

Задание 7 (про абитуриентов)

In [ ]:
'''Ваш ответ на задание №7
- верно указана применяемая статистика и обоснование ее применения
- получено расчетное значение статистики
- получено критическое значение статистики
- получено значение p_value
- сделан статистический вывод на основе сравнения критического значения с расчетным
- сделан статистический вывод на основе сравнения p-value и уровня значимости
- сделан содержательный вывод по поставленной проблеме'''

In [13]:
# Задача 7
# Дано: 10 абитуриентов прошли тест до и после подготовительных курсов.
# Данные (до, после), уровень значимости α=0.01
before = [7, 6, 5, 4, 6, 7, 8, 9, 1, 2]
after =  [9, 6, 4, 5, 7, 2, 10,3, 8, 5]

# Нулевая гипотеза H0: средние значения до и после равны.
# Альтернативная H1: средние значения отличаются.
# Применяем парный t-критерий, так как сравниваем связанные выборки (одни и те же абитуриенты).

import math
from scipy.stats import t, ttest_rel

alpha = 0.01
t_stat, p_value = ttest_rel(before, after)

# Число степеней свободы: n-1 = 10-1 = 9
df = len(before) - 1
# Критическое значение t для двустороннего теста при α=0.01
t_crit = t.ppf(1 - alpha/2, df)

print("Задача 7")
print("Применяемая статистика: Парный t-тест (ttest_rel).")
print("t-статистика:", t_stat)
print("Степени свободы:", df)
print("Критическое значение t (α=0.01): ±", round(t_crit, 3))
print("p-value:", p_value)

# Сравнение с критическим значением
if abs(t_stat) > t_crit:
    print("Отвергаем H0 на основании критического значения.")
else:
    print("Нет оснований отвергать H0 по критическому значению.")

# Сравнение p-value и α
if p_value < alpha:
    print("Отвергаем H0 на основании p-value.")
else:
    print("Нет оснований отвергать H0 по p-value.")

# Содержательный вывод
if p_value < alpha:
    print("Содержательный вывод: Подготовительные курсы статистически значимо повлияли на результаты.")
else:
    print("Содержательный вывод: Статистических доказательств влияния курсов нет при уровне значимости 0.01.")



Задача 7
Применяемая статистика: Парный t-тест (ttest_rel).
t-статистика: -0.33488734331156855
Степени свободы: 9
Критическое значение t (α=0.01): ± 3.25
p-value: 0.745383070238066
Нет оснований отвергать H0 по критическому значению.
Нет оснований отвергать H0 по p-value.
Содержательный вывод: Статистических доказательств влияния курсов нет при уровне значимости 0.01.


*  Применяемая статистика: Парный t-тест (ttest_rel).
*  t-статистика: -0.33488734331156855
*  Степени свободы: 9
*  Критическое значение t (α=0.01): ± 3.25
*  p-value: 0.745383070238066
*  Нет оснований отвергать H0 по критическому значению.
*  Нет оснований отвергать H0 по p-value.
*  Содержательный вывод: Статистических доказательств влияния курсов нет при уровне значимости 0.01.

Задание 8 (про фрукты и IQ)

In [ ]:
'''Ваш ответ на задание №8
- верно указана применяемая статистика и обоснование ее применения
- получено расчетное значение статистики
- получено критическое значение статистики
- получено значение p_value
- сделан статистический вывод на основе сравнения критического значения с расчетным
- сделан статистический вывод на основе сравнения p-value и уровня значимости
- сделан содержательный вывод по поставленной проблеме'''

In [14]:
# Задача 8
# Дано две выборки IQ для двух групп:
# 1-я группа (Апельсины): 125,138,112,130,124,125,110,112,117,126
# 2-я группа (Груши):    124,102,142,130,131,124,127,115,108,109
# Проверить гипотезу о равенстве дисперсий при α=0.05.

oranges = [125,138,112,130,124,125,110,112,117,126]
pears   = [124,102,142,130,131,124,127,115,108,109]

alpha = 0.05
n1, n2 = len(oranges), len(pears)
mean_o = sum(oranges)/n1
mean_p = sum(pears)/n2

var_o = sum((x - mean_o)**2 for x in oranges)/(n1-1)
var_p = sum((y - mean_p)**2 for y in pears)/(n2-1)

# Нулевая гипотеза H0: σ1^2 = σ2^2
# Альтернативная H1: σ1^2 ≠ σ2^2
# Используем F-тест: F = max(var_o, var_p) / min(var_o, var_p)

if var_o > var_p:
    F = var_o / var_p
    df1, df2 = n1-1, n2-1
else:
    F = var_p / var_o
    df1, df2 = n2-1, n1-1

from scipy.stats import f

# Критические значения F для двустороннего теста:
# При уровне значимости α=0.05, учитываем двухсторонний тест:
# F_crit_lower = f.ppf(alpha/2, df1, df2)
# F_crit_upper = f.ppf(1 - alpha/2, df1, df2)

F_crit_lower = f.ppf(alpha/2, df1, df2)
F_crit_upper = f.ppf(1 - alpha/2, df1, df2)

# p-value для двустороннего теста:
# p_one_side = P(F > observed) или P(F < observed), для двухстороннего умножаем на 2 при симметрии
# Но так как распределение F не симметрично, p-value считаем напрямую:
# p-value = 2 * min(P(F < F_observed), P(F > F_observed))
p_lower = f.cdf(F, df1, df2)
p_upper = 1 - p_lower
p_value = 2 * min(p_lower, p_upper)

print("\nЗадача 8")
print("Применяемая статистика: F-тест для сравнения дисперсий.")
print("F-статистика:", F)
print("df1:", df1, "df2:", df2)
print("Критическое значение F нижнее:", F_crit_lower, "верхнее:", F_crit_upper)
print("p-value:", p_value)

# Сравнение с критическими значениями
if F < F_crit_lower or F > F_crit_upper:
    print("Отвергаем H0 на основании критических значений F.")
else:
    print("Нет оснований отвергать H0 по критическому значению.")

# Сравнение p-value с α
if p_value < alpha:
    print("Отвергаем H0 на основании p-value.")
else:
    print("Нет оснований отвергать H0 по p-value.")

# Содержательный вывод
if p_value < alpha:
    print("Содержательный вывод: Дисперсии статистически значимо различаются.")
else:
    print("Содержательный вывод: Нет статистически значимых доказательств различия дисперсий при α=0.05.")


Задача 8
Применяемая статистика: F-тест для сравнения дисперсий.
F-статистика: 1.9061769156692805
df1: 9 df2: 9
Критическое значение F нижнее: 0.24838585469445493 верхнее: 4.025994158282978
p-value: 0.3505872411644839
Нет оснований отвергать H0 по критическому значению.
Нет оснований отвергать H0 по p-value.
Содержательный вывод: Нет статистически значимых доказательств различия дисперсий при α=0.05.


*  Применяемая статистика: F-тест для сравнения дисперсий.
*  F-статистика: 1.9061769156692805
*  df1: 9 df2: 9
*  Критическое значение F нижнее: 0.24838585469445493 верхнее: 4.025994158282978
*  p-value: 0.3505872411644839
*  Нет оснований отвергать H0 по критическому значению.
*  Нет оснований отвергать H0 по p-value.
*  Содержательный вывод: Нет статистически значимых доказательств различия дисперсий при α=0.05.

Задание 9 (про самолеты)

In [ ]:
'''Ваш ответ на задание №9
- верно указана применяемая статистика и обоснование ее применения
- получено расчетное значение статистики
- получено критическое значение статистики
- получено значение p_value
- сделан статистический вывод на основе сравнения критического значения с расчетным
- сделан статистический вывод на основе сравнения p-value и уровня значимости
- сделан содержательный вывод по поставленной проблеме'''

In [15]:
# Задача 9
# Дано распределение количества вылетов:
# Вылеты:  1   2   3   4   5   6   7   8   9
# Кол-во: 36  23  11  13   5   3   2   3   4 (итого 100)
# Предложим распределение Пуассона с λ, равной среднему числу вылетов.

data = [36, 23, 11, 13, 5, 3, 2, 3, 4]
k_values = [1,2,3,4,5,6,7,8,9]
N = sum(data)  # должно быть 100

mean = sum(k*freq for k, freq in zip(k_values, data))/N

# Предполагаем Poisson(mean)
# Найдем ожидаемые частоты:
from math import exp, factorial
from scipy.stats import chi2

def poisson_pmf(k, lam):
    return (lam**k * exp(-lam))/math.factorial(k)

expected = [N * poisson_pmf(k, mean) for k in k_values]

# Для применения критерия χ2 желательно, чтобы ожидаемые частоты были >=5.
# Если где-то меньше, объединяем категории. Для упрощения предположим,
# что мы можем объединить хвостовые категории.
# Проверим минимальные ожидаемые значения:
# Если некоторые ожидаемые менее 5, объединим последние категории.

while any(e < 5 for e in expected) and len(expected) > 1:
    # Объединяем последние две категории
    expected[-2] += expected[-1]
    data[-2] += data[-1]
    expected.pop()
    data.pop()
    k_values.pop()

# Теперь у нас есть новая группировка данных.
# Считаем статистику χ2:
chi_stat = sum((obs - exp)**2/exp for obs, exp in zip(data, expected))

# Число степеней свободы: df = (число категорий после объединения) - 1 - (число оцененных параметров)
# Здесь мы оценили λ по выборке, значит потеря 1 степени свободы.
df = len(data) - 1 - 1

alpha = 0.05
chi_crit = chi2.ppf(1 - alpha, df)

# p-value
p_value = 1 - chi2.cdf(chi_stat, df)

print("\nЗадача 9")
print("Применяемая статистика: χ2-критерий согласия для распределения Пуассона.")
print("χ2-статистика:", chi_stat)
print("Степени свободы:", df)
print("Критическое значение χ2 (α=0.05):", chi_crit)
print("p-value:", p_value)

if chi_stat > chi_crit:
    print("Отвергаем H0 на основании критического значения χ2.")
else:
    print("Нет оснований отвергать H0 по критическому значению.")

if p_value < alpha:
    print("Отвергаем H0 на основании p-value.")
else:
    print("Нет оснований отвергать H0 по p-value.")

# Содержательный вывод
if p_value < alpha:
    print("Содержательный вывод: Предложенное распределение (Пуассона) не подходит к данным при α=0.05.")
else:
    print("Содержательный вывод: Предложенное распределение (Пуассона) согласуется с данными на уровне доверия 0.95.")


Задача 9
Применяемая статистика: χ2-критерий согласия для распределения Пуассона.
χ2-статистика: 34.71819778557874
Степени свободы: 4
Критическое значение χ2 (α=0.05): 9.487729036781154
p-value: 5.307506091289937e-07
Отвергаем H0 на основании критического значения χ2.
Отвергаем H0 на основании p-value.
Содержательный вывод: Предложенное распределение (Пуассона) не подходит к данным при α=0.05.


* Применяемая статистика: χ2-критерий согласия для распределения Пуассона.
* χ2-статистика: 34.71819778557874
* Степени свободы: 4
* Критическое значение χ2 (α=0.05): 9.487729036781154
* p-value: 5.307506091289937e-07
* Отвергаем H0 на основании критического значения χ2.
* Отвергаем H0 на основании p-value.
* Содержательный вывод: Предложенное распределение (Пуассона) не подходит к данным при α=0.05.

Задание 10 (про цвет глаз и волос)

In [ ]:
'''Ваш ответ на задание №10
- верно указана применяемая статистика и обоснование ее применения
- получено расчетное значение статистики
- получено критическое значение статистики
- получено значение p_value
- сделан статистический вывод на основе сравнения критического значения с расчетным
- сделан статистический вывод на основе сравнения p-value и уровня значимости
- сделан содержательный вывод по поставленной проблеме'''

In [16]:
# Задача 10
# Исследуется связь между цветом глаз (Голубые, Карие, Зеленые) и цветом волос (Светлые, Темные, Рыжие).
# Данные:
#              Светлые  Темные  Рыжие
# Голубые         35      52      13
# Карие           28      42      10
# Зеленые          7      10       3

# Всего: 3 строки (цвет глаз), 3 столбца (цвет волос).
# Проверяем гипотезу о независимости признаков:
# H0: Цвет волос и цвет глаз независимы.
# H1: Есть связь между цветом волос и цветом глаз.
# Уровень доверия 0.95 => α=0.05.

import numpy as np
from scipy.stats import chi2_contingency, chi2

# Таблица сопряженности
obs = np.array([[35, 52, 13],
                [28, 42, 10],
                [ 7, 10,  3]])

chi_stat, p_value, dof, expected = chi2_contingency(obs)

alpha = 0.05
chi_crit = chi2.ppf(1 - alpha, dof)

print("Задача 10")
print("Применяемая статистика: χ2-критерий согласия (проверка независимости).")
print("Расчетная χ2-статистика:", chi_stat)
print("Степени свободы:", dof)
print("Критическое значение χ2 (α=0.05):", round(chi_crit, 3))
print("p-value:", p_value)

# Статистический вывод на основании критического значения
if chi_stat > chi_crit:
    print("Отвергаем H0 на основании сравнения с критическим значением χ2.")
else:
    print("Нет оснований отвергать H0 по критическому значению χ2.")

# Статистический вывод на основании p-value
if p_value < alpha:
    print("Отвергаем H0 на основании p-value.")
else:
    print("Нет оснований отвергать H0 по p-value.")

# Содержательный вывод
if p_value < alpha:
    print("Содержательный вывод: Существует статистически значимая связь между цветом волос и цветом глаз.")
else:
    print("Содержательный вывод: Статистически значимых доказательств связи между цветом волос и цветом глаз не обнаружено при уровне доверия 0.95.")

Задача 10
Применяемая статистика: χ2-критерий согласия (проверка независимости).
Расчетная χ2-статистика: 0.09615384615384615
Степени свободы: 4
Критическое значение χ2 (α=0.05): 9.488
p-value: 0.9988806869095861
Нет оснований отвергать H0 по критическому значению χ2.
Нет оснований отвергать H0 по p-value.
Содержательный вывод: Статистически значимых доказательств связи между цветом волос и цветом глаз не обнаружено при уровне доверия 0.95.


* Применяемая статистика: χ2-критерий согласия (проверка независимости).
* Расчетная χ2-статистика: 0.09615384615384615
* Степени свободы: 4
* Критическое значение χ2 (α=0.05): 9.488
* p-value: 0.9988806869095861
* Нет оснований отвергать H0 по критическому значению χ2.
* Нет оснований отвергать H0 по p-value.
* Содержательный вывод: Статистически значимых доказательств связи между цветом волос и цветом глаз не обнаружено при уровне доверия 0.95.


Задание 11 (про тигров)

In [ ]:
'''Ваш ответ на задание №11
- верно указана применяемая статистика и обоснование ее применения
- получено расчетное значение статистики
- получено значение p_value
- сделан статистический вывод на основе сравнения p-value и уровня значимости
- в случае, если нулевая гипотеза отвергнута, сделан тест Тьюки для подробного сравнения средних
- сделано дополнение к статистическому выводу по результатам теста Тьюки
- сделан содержательный вывод по поставленной проблеме'''

In [17]:
# Задача 11
# Даны данные о массе тигров при разных типах кормов (например, 4 группы кормов, данные по каждой группе).
# Нужно провести однофакторный дисперсионный анализ (ANOVA) для проверки равенства средних значений.
# H0: Средние значения во всех группах одинаковы.
# H1: Хотя бы в одной группе среднее отличается.
# Если нулевая гипотеза отвергается, применим тест Тьюки для парных сравнений.

import numpy as np
from scipy.stats import f_oneway
import statsmodels.api as sm
from statsmodels.stats.multicomp import pairwise_tukeyhsd

# Допустим, у нас 4 группы (№1, №2, №3, №4) по данным из задания.
# Данные (примерные, заменить на фактические, если в задании даны конкретные числа):
group1 = [151,135,137,118,132,135,131,121,137,140,152,133,151,132]
group2 = [108,123,145,103,110,124,118,106,105,114,108]
group3 = [140,152,149,132,124,146,145,147,139,157,139]
group4 = [127,135,110,116,119,123,116,134,121,109]

alpha = 0.05

# ANOVA
f_stat, p_value = f_oneway(group1, group2, group3, group4)

print("Задача 11")
print("Применяемая статистика: Однофакторный ANOVA F-тест.")
print("F-статистика:", f_stat)
print("p-value:", p_value)

# Вывод на основании p-value
if p_value < alpha:
    print("Отвергаем H0: есть статистически значимые различия между средними массами для разных типов кормов.")
    # Применяем тест Тьюки для определения пар, в которых есть различия.
    # Подготовим данные для Тьюки.
    data = group1 + group2 + group3 + group4
    labels = (["Group1"]*len(group1) + ["Group2"]*len(group2) +
              ["Group3"]*len(group3) + ["Group4"]*len(group4))
    tukey_result = pairwise_tukeyhsd(endog=data, groups=labels, alpha=alpha)
    print("\nРезультаты теста Тьюки:")
    print(tukey_result)

    # Содержательный вывод:
    # Если некоторые пары групп различаются по тесту Тьюки,
    # значит для тех типов кормов средние массы тигров статистически значимо отличаются.
    print("Содержательный вывод: Определены специфические группы кормов, между которыми наблюдается значимая разница в средней массе тигров.")
else:
    print("Нет оснований отвергать H0: статистически значимых различий между средними не обнаружено.")
    print("Содержательный вывод: Тип корма не оказывает статистически значимого влияния на среднюю массу тигров при уровне значимости 0.05.")

Задача 11
Применяемая статистика: Однофакторный ANOVA F-тест.
F-статистика: 17.66446410174202
p-value: 1.4413310817011847e-07
Отвергаем H0: есть статистически значимые различия между средними массами для разных типов кормов.

Результаты теста Тьюки:
 Multiple Comparison of Means - Tukey HSD, FWER=0.05  
group1 group2 meandiff p-adj   lower    upper   reject
------------------------------------------------------
Group1 Group2 -21.1623    0.0 -32.2337 -10.0909   True
Group1 Group3   6.6558 0.3852  -4.4156  17.7272  False
Group1 Group4 -15.0714 0.0052 -26.4486  -3.6942   True
Group2 Group3  27.8182    0.0  16.1013  39.5351   True
Group2 Group4   6.0909 0.5327  -5.9153  18.0971  False
Group3 Group4 -21.7273 0.0001 -33.7335  -9.7211   True
------------------------------------------------------
Содержательный вывод: Определены специфические группы кормов, между которыми наблюдается значимая разница в средней массе тигров.


* Применяемая статистика: Однофакторный ANOVA F-тест.
* F-статистика: 17.66446410174202
* p-value: 1.4413310817011847e-07
* Отвергаем H0: есть статистически значимые различия между средними массами для разных типов кормов.

* Результаты теста Тьюки:
*  Multiple Comparison of Means - Tukey HSD, FWER=0.05  
* ======================================================
* group1 group2 meandiff p-adj   lower    upper   reject
------------------------------------------------------
* Group1 Group2 -21.1623    0.0 -32.2337 -10.0909   True
* Group1 Group3   6.6558 0.3852  -4.4156  17.7272  False
* Group1 Group4 -15.0714 0.0052 -26.4486  -3.6942   True
* Group2 Group3  27.8182    0.0  16.1013  39.5351   True
* Group2 Group4   6.0909 0.5327  -5.9153  18.0971  False
* Group3 Group4 -21.7273 0.0001 -33.7335  -9.7211   True
------------------------------------------------------
* Содержательный вывод: Определены специфические группы кормов, между которыми наблюдается значимая разница в средней массе тигров.

Задание 12 (про белок)

In [ ]:
'''Ваш ответ на задание №12
- верно указана применяемая статистика и обоснование ее применения
- расчет выполнен через последовательные суммы квадратов
- получены расчетные значения статистики
- получены значения p_value
- сделан статистический вывод на основе сравнения p-value и уровня значимости
- сделан содержательный вывод по поставленной проблеме'''

In [18]:
# Задача 12
# Изучается влияние размножения (фактор: с выводком / без выводка) и возраста (фактор: 1 год, 2 года, 3 и более лет) на массу тела.
# Имеются данные по факторам, нужно выполнить двухфакторный дисперсионный анализ (ANOVA) с использованием последовательных сумм квадратов.
# H0: Нет влияния факторов или взаимодействия на среднюю массу (все средние равны).
# H1: Есть влияние хотя бы одного фактора или взаимодействия.

# Допустим, данные заданы в таблице:
# Возраст: 1 год, 2 года, 3 и более лет
# Фактор размножения: Без выводка / С выводком
# Данные (примерные, необходимо подставить фактические значения из задания):
# Без выводка, 1 год: [440,438,429,502,602]
# Без выводка, 2 года: [892,868,855,866,932]
# Без выводка, 3+ лет: [1575,849,759,1602,1327]

# С выводком, 1 год: [308,328,326,326,325]
# С выводком, 2 года: [737,798,876,810,861]
# С выводком, 3+ лет: [1001,901,958,1032,883]

import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols

data = {
    'mass': [440,438,429,502,602,  # Без выводка, 1 год
             892,868,855,866,932,  # Без выводка, 2 года
             1575,849,759,1602,1327, # Без выводка, 3+ лет
             308,328,326,326,325,  # С выводком, 1 год
             737,798,876,810,861,  # С выводком, 2 года
             1001,901,958,1032,883], # С выводком, 3+ лет
    'reproduction': (["Без выводка"]*5 + ["Без выводка"]*5 + ["Без выводка"]*5 +
                     ["С выводком"]*5 + ["С выводком"]*5 + ["С выводком"]*5),
    'age': (["1 год"]*5 + ["2 года"]*5 + ["3+ лет"]*5 +
            ["1 год"]*5 + ["2 года"]*5 + ["3+ лет"]*5)
}

df = pd.DataFrame(data)

# Двухфакторный ANOVA:
# Модель: mass ~ C(reproduction) + C(age) + C(reproduction):C(age)
model = ols('mass ~ C(reproduction)*C(age)', data=df).fit()
anova_table = sm.stats.anova_lm(model, typ=2)  # typ=2 для последовательных сумм квадратов

alpha = 0.05

print("Задача 12")
print("Применяемая статистика: двухфакторный дисперсионный анализ (ANOVA).")
print(anova_table)

# Из таблицы ANOVA берем значения F-статистик и p-value для факторов и взаимодействия:
# anova_table содержит строки для C(reproduction), C(age), и C(reproduction):C(age).
# Проверяем p-value для каждого фактора и взаимодействия.

for factor in anova_table.index:
    if factor != 'Residual':
        F_stat = anova_table.loc[factor, 'F']
        p_value = anova_table.loc[factor, 'PR(>F)']
        print(f"\nФактор: {factor}")
        print("F-статистика:", F_stat)
        print("p-value:", p_value)

        if p_value < alpha:
            print("Отвергаем H0 для фактора/взаимодействия на основании p-value.")
        else:
            print("Нет оснований отвергать H0 для данного фактора/взаимодействия по p-value.")

# Содержательный вывод:
# Если для какого-либо фактора или взаимодействия p_value < α,
# значит этот фактор или взаимодействие статистически значимо влияет на целевой показатель (массу тела).
print("\nСодержательный вывод:")
significant_factors = [f for f in anova_table.index if f != 'Residual' and anova_table.loc[f, 'PR(>F)'] < alpha]

if significant_factors:
    print("На уровне доверия 0.95 найдены факторы/взаимодействия, статистически значимо влияющие на массу тела.")
    print("Дальнейший анализ может указать, как именно меняется масса при разных категориях факторов.")
else:
    print("Нет статистически значимого влияния ни со стороны факторов, ни со стороны их взаимодействия на уровне доверия 0.95.")

Задача 12
Применяемая статистика: двухфакторный дисперсионный анализ (ANOVA).
                           sum_sq    df          F        PR(>F)
C(reproduction)          202705.2   1.0   7.078626  1.368668e-02
C(age)                  2427075.8   2.0  42.377707  1.333903e-08
C(reproduction):C(age)    50688.2   2.0   0.885036  4.257428e-01
Residual                 687269.6  24.0        NaN           NaN

Фактор: C(reproduction)
F-статистика: 7.07862649533747
p-value: 0.01368667957570527
Отвергаем H0 для фактора/взаимодействия на основании p-value.

Фактор: C(age)
F-статистика: 42.37770679803103
p-value: 1.3339031198372199e-08
Отвергаем H0 для фактора/взаимодействия на основании p-value.

Фактор: C(reproduction):C(age)
F-статистика: 0.8850360906404267
p-value: 0.42574276908860065
Нет оснований отвергать H0 для данного фактора/взаимодействия по p-value.

Содержательный вывод:
На уровне доверия 0.95 найдены факторы/взаимодействия, статистически значимо влияющие на массу тела.
Дальнейший анали


Задача 12

* Применяемая статистика: двухфакторный дисперсионный анализ (ANOVA).

* Фактор: C(reproduction)
* F-статистика: 7.07862649533747
* p-value: 0.01368667957570527
*  Отвергаем H0 для фактора/взаимодействия на основании p-value.

* Фактор: C(age)
* F-статистика: 42.37770679803103
* p-value: 1.3339031198372199e-08
* Отвергаем H0 для фактора/взаимодействия на основании p-value.

* Фактор: C(reproduction):C(age)
* F-статистика: 0.8850360906404267
* p-value: 0.42574276908860065
* Нет оснований отвергать H0 для данного фактора/взаимодействия по p-value.

Содержательный вывод:
На уровне доверия 0.95 найдены факторы/взаимодействия, статистически значимо влияющие на массу тела.
Дальнейший анализ может указать, как именно меняется масса при разных категориях факторов.